Jellyfin.Server/Extensions/ApiServiceCollectionExtensions.cs

```diff
@@ -169,36 +171,104 @@ public static AuthenticationBuilder AddCustomAuthentication(this IServiceCollect
                 .AddScheme<AuthenticationSchemeOptions, CustomAuthenticationHandler>(AuthenticationSchemes.CustomAuthentication, null);
         }
 
+        /// <summary>
+        /// Parses a string array of IPAddresses/Hosts, and executing <paramref name="callback"/> on each valid ip address.
+        /// </summary>
+        /// <param name="config">The <see cref="NetworkConfiguration"/> containing the config settings.</param>
+        /// <param name="userList">The string array to parse.</param>
+        /// <param name="callback">The callback method.</param>
+        public static void ParseList(NetworkConfiguration config, string[] userList, Action<IPAddress, int> callback)
+        {
+            for (var i = 0; i < userList.Length; i++)
+            {
+                if (IPNetAddress.TryParse(userList[i], out var addr))
+                {
+                    if ((!config.EnableIPV4 && addr.AddressFamily == AddressFamily.InterNetwork)
+                         || (!config.EnableIPV6 && addr.AddressFamily == AddressFamily.InterNetworkV6))
+                    {
+                        continue;
+                    }
+
+                    if (config.EnableIPV6 && config.EnableIPV4 && addr.AddressFamily == AddressFamily.InterNetwork)
+                    {
+                        // If the server is using dual-mode sockets, IPv4 addresses are supplied in an IPv6 format.
+                        // https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/proxy-load-balancer?view=aspnetcore-5.0 .
+                        callback(addr.Address.MapToIPv6(), addr.PrefixLength);
+                    }
+                    else
+                    {
+                        callback(addr.Address, addr.PrefixLength);
+                    }
+                }
+                else if (IPHost.TryParse(userList[i], out var host))
```

Comment:
Duplicated code basically. I also dislike the use of "callback".

Reply:
1st one is an IP network address parser, second one is an Host parser.

Reply:
renamed to addressDelegate

Reply:
I was referring to the use of a delegate, not the name. Sorry for being unclear.

Reply:
Didn't use a sub-function at first - but that took away the "cleanness" of what the Configure<ForwardedHeadersOptions> was trying to achieve, as well as making the method pretty large.

The problem is that dotnet expects IP addresses to be stored in KnownProxies in a multitude of ways depending if IP6 is enabled or not (Don't know why the couldn't do this internally).

KnownNetworks, however expect IPNetwork entries so it's not a one size fits all.

Using delegates seemed to be the cleanest method. The only other alternative i can think of is to pass options as a parameter to the function, along with a value of "1" for KnownProxies, or "2" for KnownNetworks.

Reply:
Removed delegate.